tests/test_causal_graph.py

```diff
@@ -538,6 +538,25 @@ def test_adjacency_consistency(self):
 
         self.assertEqual(CausalGraph.from_adjacency_matrix(*cg.to_numpy()), cg)
 
+    def test_copy(self):
+        # test copy preserves the correct class
+        ts_cg = TimeSeriesCausalGraph()
```

Comment:
Do `ts_cg.add_node('a', meta={'some': 'thing'})` just so meta is sitting on node too.